pkg/controller/atlas/manager.go

```diff
@@ -0,0 +1,79 @@
+package atlas
+
+import (
+	"net/url"
+	"strings"
+
+	"go.mongodb.org/atlas/mongodbatlas"
+	"go.uber.org/zap"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	mdbv1 "github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/util/httputil"
+)
+
+const govAtlasDomain = "mongodbgov.com"
+
+type Manager interface {
+	CreateConnection(secretRef *client.ObjectKey, log *zap.SugaredLogger) (Connection, error)
+	CreateClient(connection *Connection, log *zap.SugaredLogger, opts ...httputil.ClientOpt) (mongodbatlas.Client, error)
+	IsCloudGov() bool
+	IsResourceSupported(resource mdbv1.AtlasCustomResource) bool
+}
+
+type CloudManager struct {
+	k8sClient       client.Client
+	domain          string
+	globalSecretRef client.ObjectKey
+}
+
+func NewCloudFactory(atlasDomain string, globalSecretRef client.ObjectKey, k8sClient client.Client) *CloudManager {
```

Comment:
What is this Java code? ;-p 

I would just call it `NewProductionCloud` and returns a `ProductionCloud` implementing the `Cloud` interface.